### PR TITLE
[GSoC] Added Bubble Scroll

### PR DESCRIPTION
**Pull request by Ayan-10**
**Tasks**

- [x] Add date with Image Entity
- [x] Add bubble scroll bar showing the date

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageFileLoader.kt [hunk 1]
```diff
@@ -84,7 +93,22 @@ class ImageFileLoader(val context: Context) : CoroutineScope{
                 if (file != null && file.exists()) {
                     if (name != null && path != null && bucketName != null) {
                         val uri = ContentUris.withAppendedId(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, id)
-                        val image = Image(id, name, uri, path, bucketId, bucketName)
+
+                        val calendar = Calendar.getInstance()
+                        calendar.timeInMillis = date * 1000L
+                        val date: Date = calendar.time
+                        val dateFormat = SimpleDateFormat("MMM dd, yyyy", Locale.getDefault())
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageFileLoader.kt:100]: It looks very strange in Japanese (`7月 21, 2022`) as Japanese dates always follow a different formatting from English. How about letting the locale do the formatting?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Wonderful, it works great!
And it almost does not change the size of the APK.

I just added a few minor asks.

File: app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageFileLoader.kt (hunk 1 above)
Comment (Ayan-10) [reply app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageFileLoader.kt]: @nicolas-raoul Is it ok now?

**Comment by nicolas-raoul** [comment]
I just noticed that it shows the file's date rather than the EXIF date. Getting the EXIF date by parsing all files would take too much time, but by any chance do we have it somewhere already? Maybe not.

**Review by nicolas-raoul — APPROVED** [review]

**Comment by Ayan-10** [comment]
@nicolas-raoul `exif.getAttribute(ExifInterface.TAG_DATETIME)` are you talking about this? I have tried once to get the date with this, but it's returning null every time. I will surely try again later to get the EXIF date.

**Pull request merged**